lib/ansible/plugins/action/__init__.py

```diff
@@ -1090,3 +1136,139 @@ def _find_needle(self, dirname, needle):
+            if self._connection.has_pipelining:
+                res = self._low_level_execute_command(found_interpreters[0], sudoable=False, in_data=platform_script)
+            else:
+                # FIXME: implement on-disk case (via script action or ?)
+                raise NotImplementedError('pipelining required for extended interpreter discovery')
+
+            platform_info = json.loads(res.get('stdout'))
+
+            distro = platform_info.get('distro')
+            version = platform_info.get('version')
+
+            if not distro or not version:
+                raise NotImplementedError('unable to get Linux distribution/version info')
+
+            version_map = platform_python_map.get(distro.lower().strip())
+            if not version_map:
+                raise NotImplementedError('unsupported Linux distribution: {0}'.format(distro))
+
+            platform_interpreter = self._version_fuzzy_match(version, version_map)
+
+            # provide a transition period for hosts that were using /usr/bin/python previously (but shouldn't have been)
+            if discovery_mode == 'auto_legacy':
+                if platform_interpreter != '/usr/bin/python' and '/usr/bin/python' in found_interpreters:
+                    # FIXME: support comments in sivel's deprecation scanner so we can get reminded on this
+                    self._discovery_deprecation_warnings.append(dict(
+                        msg="Distribution {0} {1} should use {2}, but is using "
+                            "/usr/bin/python for backward compatibility with prior Ansible releases. "
+                            "A future Ansible release will default to using the discovered platform "
+                            "python for this host. See [FIXME DOC LINK TBD] for more information"
+                        .format(distro, version, platform_interpreter),
+                        version='2.12'))
+                    return '/usr/bin/python'
+
+            if platform_interpreter not in found_interpreters:
+                if platform_interpreter not in bootstrap_python_list:
+                    # sanity check to make sure we looked for it
+                    self._discovery_warnings \
+                        .append("Platform interpreter {0} is missing from bootstrap list"
+                                .format(platform_interpreter))
+
+                if discovery_mode != 'auto_silent':
+                    self._discovery_warnings\
+                        .append("Distribution {0} {1} should use {2}, but is using {3}, since the "
+                                "discovered platform python interpreter was not present. See [FIXME DOC LINK TBD] "
+                                "for more information."
+                                .format(distro, version, platform_interpreter, found_interpreters[0]))
+                return found_interpreters[0]
+
+            return platform_interpreter
+        except NotImplementedError as ex:
```

Comment:
Should probably also just catch Exception in another `except`, so fallback will always occur (but we log the unexpected ones more aggressively and have a different warning).